_inc/client/components/jetpack-notices/index.jsx

```diff
@@ -141,30 +142,29 @@ export const DevModeNotice = React.createClass( {
 	displayName: 'DevModeNotice',
 
 	render() {
-		const devMode = window.Initial_State.connectionStatus.devMode;
-		if ( devMode.isActive ) {
-			if (devMode.filter) {
-				const text = __('Currently in {{a}}Development Mode{{/a}} via the jetpack_development_mode filter.{br/}}Some features are disabled.',
+		if ( getSiteConnectionStatus( this.props ) === 'dev' ) {
```

Comment:
Could it not be possible to have multiple ways to put your site into dev mode.
maybe an approach where we display all the items in 1 string would work best?